app/controllers/api/oceans_controller.rb

```diff
@@ -1,15 +1,26 @@
 class Api::OceansController < ApplicationController
   def index
-    params = Rack::Utils.parse_query URI(request.url).query
-    count = params['count']
-    count = '4' if count.nil? || count.empty?
-    if count.to_i.negative? || count.to_i > 20
-      render nothing: true, status: :bad_request
+    if !limit_valid?
+      render_bad_request
     else
-      oceans = Array.new(count.to_i) do
+      oceans = Array.new(number_of_oceans) do
         OceanService.new(Random.rand(1..5)).call
       end
       render json: oceans
     end
```

Comment:
nice, the only thing that is missing now is the before_action thingy http://guides.rubyonrails.org/action_controller_overview.html#filters 

A filter also terminates the request if there is a render or redirect call happenning in there.